### DISCUSSION TITLE
MST model to JSON Schema generator

### DISCUSSION THREAD

**Discussion by beepsoft** [mobxjs/mobx-state-tree#1858] [Q&A]
I wonder if I have an MST model (generated using mst-gql from a Graphql Schema), is it possible to generate a JSON Schema based on just the runtime types of the MST model properties? Does any such tool exists?

**Comment by stoefln** [comment]
Would also be interested in how to transform the type information to other formats (like TS).

**Reply by beepsoft** [reply]
As the MST model is generated based on the graphql schema in case of mst-gql, maybe graphql-2-json-schema can be used for this https://www.npmjs.com/package/graphql-2-json-schema But in a pure MST based model the question is still relevant, I think.
